### PR TITLE
chore: use int instead of unsigned int in comparison operation

### PR DESCRIPTION
**Pull request by adit4443ya**
Towards [bug fix](https://github.com/lfortran/lfortran/pull/5011#discussion_r1821262606)
in continuation of #5011

### CODE REVIEW

File: src/lfortran/semantics/ast_body_visitor.cpp [hunk 1]
```diff
@@ -3447,7 +3447,7 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 do_concurrent->m_head = do_concurrent_head.p;
                 do_concurrent->n_head = do_concurrent_head.size();
                 tmp = (ASR::asr_t*) do_concurrent;
-            } else if (openmp_collapse == true && !omp_constructs.empty() && collapse_value>loop_nesting-1-omp_constructs.size()+1) {
+            } else if (openmp_collapse == true && !omp_constructs.empty() && collapse_value > loop_nesting - 1 - static_cast<int>(omp_constructs.size()) + 1) {
```
Comment (gxyd) [inline src/lfortran/semantics/ast_body_visitor.cpp:3450]: Two considerations/questions:
1. maybe wrap the condition after `&&` over to the next line?
2. any reason to have non-simplified `-1` and `+1` on the rhs of the greater than comparison operator?

**Comment by Pranavchiku** [comment]
Let's make a practice of having meaningful commit messages and PR names, this will avoid squash and merge in PRs and also help in preserving commit history.

**Review by certik — APPROVED** [review]
I think this is the right fix. Thanks!

@gxyd feedback can be addressed in another PR if needed, I think it's minor.

**Comment by certik** [comment]
I missed that "small fix" commit log. As @Pranavchiku said, please write a meaningful commit log messages. Instead of "small fix", write what was fixed and why.

**Pull request merged**